### PR TITLE
Prevent MVT tile downloads when layer not visible 

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6747

#### Change List
- Prevent tiles being fetched when `visible=false`
- Test

Note that the tileJSON will still be fetched if the `data` prop is a URL. This is intentional.

### CODE REVIEW

File: modules/test-utils/src/lifecycle-test.js [hunk 1]
```diff
@@ -247,6 +249,9 @@ function update({layerManager, deckRenderer}) {
         });
         resolve();
         return;
+      } else if (++n > 10) {
```
Comment (ibgreen) [inline modules/test-utils/src/lifecycle-test.js:252]: Use a named constant instead of 10 to make this more self documenting.

**Review by ibgreen — COMMENTED** [review]
QQ: Is this already supported by TileLayer? If not, is there a reason to make this mvt-layer specific?

**Comment by felixpalmer** [comment]
@ibgreen no, `TileLayer` suffers from the same problem. I've pushed another commit that addresses that also

**Comment by Pessimistress** [comment]
Please see discussion on the original issue.

File: modules/geo-layers/src/mvt-layer/mvt-layer.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ export default class MVTLayer extends TileLayer {
       this._updateTileData();
     }
 
-    if (this.state.data) {
+    if (this.state.data && this.props.visible) {
```
Comment (Pessimistress) [inline modules/geo-layers/src/mvt-layer/mvt-layer.js:46]: Is this necessary if `super.updateState` already checks the prop?

File: modules/geo-layers/src/tile-layer/tile-layer.js [hunk 3]
```diff
@@ -195,6 +196,9 @@ export default class TileLayer extends CompositeLayer {
   }
 
   renderLayers() {
+    if (!this.props.visible) {
+      return null;
```
Comment (Pessimistress) [inline modules/geo-layers/src/tile-layer/tile-layer.js:200]: I don't think this is right. `visible` is recommended as an approach to toggle layers without the perf impact of removing/re-adding them. No other composite layer removes its sub layers when visibility changes.

**Comment by felixpalmer** [comment]
I would say that we should abondon this PR now that https://github.com/visgl/deck.gl/pull/6754 is merged. I will close by end of week unless I hear otherwise

**Pull request closed without merging**